segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -187,4 +198,40 @@ static void delete(long ledgerId, BookKeeper bookKeeper) throws DurableDataLogEx
 
         return result;
     }
+
+    /**
+     * Gets the {@link #PROPERTY_LOG_ID} value from the given {@link LedgerHandle}, as stored in its custom metadata.
+     *
+     * @param handle The {@link LedgerHandle} to query.
+     * @return The Log Id stored in {@link #PROPERTY_LOG_ID}, or {@link #NO_LOG_ID} if not defined (i.e., due to an upgrade
+     * from a version that did not store this information).
+     */
+    static int getBookKeeperLogId(LedgerHandle handle) {
+        byte[] logIdSerialized = handle.getCustomMetadata().getOrDefault(PROPERTY_LOG_ID, null);
+        if (logIdSerialized == null) {
+            log.warn("No property '{}' found on Ledger {}. This is OK if this ledger was created prior to Pravega 0.7.1.",
+                    PROPERTY_LOG_ID, handle.getId());
+            return NO_LOG_ID;
+        }
+
+        String deserialized = null;
+        try {
+            deserialized = new String(logIdSerialized, Charsets.US_ASCII);
+            return Integer.parseInt(deserialized);
+        } catch (Exception ex) {
+            log.error("Property '{}' Ledger {} has invalid value '{}'. Returning default value.",
+                    PROPERTY_LOG_ID, handle.getId(), deserialized == null ? logIdSerialized : deserialized);
+            return NO_LOG_ID;
+        }
+    }
+
+    /**
+     * Creates a new {@link Map} that can be set as a Ledger's Custom Metadata.
+     *
+     * @param logId The {@link BookKeeperLog} Id to encode.
+     * @return An immutable {@link Map} containing the encoded Ledger's Custom Metadata.
+     */
+    private static Map<String, byte[]> createLedgerCustomMetadata(int logId) {
+        return Collections.singletonMap(PROPERTY_LOG_ID, Integer.toString(logId).getBytes(Charsets.US_ASCII));
```

Comment:
Can we also add  the stream name and a constant 'application' = 'pravega' ?

This will ease understanding the meaning of the ledger.
Especially while using a shared BK cluster
or with tools like
https://github.com/diennea/bookkeeper-visual-manager that allow to inspect a BK cluster

Reply:
A Stream is not necessarily tied to a single BK Ledger/BKLog. It is made up of many segments, and those segments are created, sealed, deleted, etc. throughout a Stream's lifecycle. We only keep BK ledgers for a short amount of time, so it doesn't make much sense from our POV to tag Stream names.

We can add Pravega as a tag too. That's not a bad idea.